liota/lib/transports/mqtt.py

```diff
@@ -165,24 +151,41 @@ def __init__(self, url, port, identity=None, tls_conf=None, qos_details=None, cl
         if self.clean_session:
             # If user passes client_id, it'll be used.  Otherwise, it is left to the underlying paho
             # to generate random client_id
-            self._paho_client = paho.Client(self.client_id, self.clean_session, self.userdata,
-                                            protocol=getattr(paho, self.protocol), transport=self.transport)
             log.info("clean_session is set to True")
         else:
             #  client_id is either auto-generated or provided by user
-            self._paho_client = paho.Client(self.client_id, self.clean_session, self.userdata,
-                                            protocol=getattr(paho, self.protocol), transport=self.transport)
             log.info("clean_session is set to False")
 
+        self._paho_client = paho.Client(self.client_id, self.clean_session, self.userdata,
+                                        protocol=getattr(paho, self.protocol), transport=self.transport)
         self._connect_result_code = sys.maxsize
         self._disconnect_result_code = sys.maxsize
         self._paho_client.on_message = self.on_message
         self._paho_client.on_publish = self.on_publish
         self._paho_client.on_subscribe = self.on_subscribe
         self._paho_client.on_connect = self.on_connect
         self._paho_client.on_disconnect = self.on_disconnect
+        self.sub_list = []
         self.connect_soc()
 
+    def on_connect(self, client, userdata, flags, rc):
+        """
+        Invoked on successful connection to a broker after connection request
+
+        :param client: The client instance for this callback
+        :param userdata: The private user data as set in Client() or userdata_set()
+        :param flags: Response flags sent by the broker
+        :param rc: The connection result
+        :return:
+        """
+        self._connect_result_code = rc
+        self._disconnect_result_code = sys.maxsize
+        log.info("Connected with result code : {0} : {1} ".format(str(rc), paho.connack_string(rc)))
+        if self.sub_list:
+            for sub_values in self.sub_list:
```

Comment:
Please change this to:

for [topic, qos, callback] in self.sub_list:

And the appropriate changes in the following lines instead of using sub_values[x].

Reply:
Done